lib/cluster.js

```diff
@@ -528,15 +531,22 @@ function workerInit() {
   };
 
   // obj is a net#Server or a dgram#Socket object.
-  cluster._getServer = function(obj, address, port, addressType, fd, cb) {
-    var message = {
-      addressType: addressType,
-      address: address,
-      port: port,
+  cluster._getServer = function(obj, options, cb) {
+    var key = [ options.address,
+                options.port,
+                options.addressType,
+                options.fd ].join(':');
+    if (indexes[key] === undefined)
+      indexes[key] = 0;
+    else
+      indexes[key]++;
+
+    var message = util._extend({
       act: 'queryServer',
-      fd: fd,
+      index: indexes[key],
```

Comment:
You can avoid the extra lookup if you write the initial one like this: `const index = indexes[key] = (indexes[key] || -1) + 1;`